File: app/src/main/java/com/sport/playsqorr/dynamic/MenuAdapter.java
package com.sport.playsqorr.dynamic;

import android.app.Activity;
import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import com.sport.playsqorr.R;
import com.sport.playsqorr.fragments.HomeFragment;

import java.util.ArrayList;

public class MenuAdapter extends RecyclerView.Adapter<MenuAdapter.MyViewHolder> {
    ArrayList<MenuData> menuArrayList;
    Context mContext;
    Activity mActivity;
    Fragment objfragment;
    int SelectedIndex = 0;

    public MenuAdapter(ArrayList<MenuData> menuArrayList, Context mContext, Activity mActivity, HomeFragment fragment) {
        this.menuArrayList = menuArrayList;
        this.mContext = mContext;
        this.mActivity = mActivity;
        this.objfragment = fragment;
    }

    public class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        public LinearLayout menuLayout;
        public TextView txtMenu;
        public ImageView imgMenuIcon;


        //        private ItemClickListener clickListener;
        public MyViewHolder(View view) {
            super(view);
            menuLayout = (LinearLayout) view.findViewById(R.id.menuLayout);
            txtMenu = (TextView) view.findViewById(R.id.txtMenu);
            imgMenuIcon = (ImageView) view.findViewById(R.id.imgMenuIcon);

        }

        public void setClickListener(AdapterView.OnItemClickListener itemClickListener) {
            this.setClickListener(itemClickListener);// = itemClickListener;
        }

        @Override
        public void onClick(View v) {
            System.out.println("testonclick");
        }
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.menu_itemview, parent, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        MenuData menulist = menuArrayList.get(position);
        holder.txtMenu.setText(menulist.getStrName());
        if (position == SelectedIndex) {
            holder.menuLayout.setBackgroundResource(R.drawable.corner_radius_black);
            holder.txtMenu.setTextColor(Color.WHITE);
            holder.imgMenuIcon.setImageResource(R.drawable.menu_selected);
        } else {
            holder.menuLayout.setBackgroundColor(Color.WHITE);
            holder.txtMenu.setTextColor(Color.BLACK);
            holder.imgMenuIcon.setImageResource(R.drawable.menu_unselected);
        }
        holder.menuLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                SelectedIndex = position;
                notifyDataSetChanged();
                if (position == 0) {
                    ((HomeFragment) objfragment).refreshSubMenu(menulist.getArrLeague());
                } else {
                    ArrayList<Leagues> leagues = new ArrayList<>();
                    Leagues objLeague = new Leagues();
                    objLeague.setLeagueAbbreviation(menulist.getStrName());
                    leagues.add(0, objLeague);
                    leagues.addAll(menulist.getArrLeague());
                    ((HomeFragment) objfragment).refreshSubMenu(leagues);
                }
            }
        });

    }

    public int getItemCount() {
        return menuArrayList.size();
    }

}

File: app/src/main/java/com/sport/playsqorr/home/ui/adapter/MenuAdapter.kt
package com.sport.playsqorr.home.ui.adapter

import com.sport.playsqorr.dynamic.MenuData
import android.app.Activity
import android.content.Context
import android.graphics.Color
import com.sport.playsqorr.home.ui.fragment.HomeFragment
import com.sport.playsqorr.SportIdMatchupTypeInterface
import androidx.recyclerview.widget.RecyclerView
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.AdapterView.OnItemClickListener
import com.sport.playsqorr.R
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import androidx.appcompat.widget.AppCompatImageView
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import com.sport.playsqorr.dynamic.Leagues
import com.sport.playsqorr.home.model.SportAndIcon
import java.util.ArrayList

class MenuAdapter(
    var menuArrayList: ArrayList<MenuData>,
    var mContext: Context,
    var mActivity: Activity,
    fragment: HomeFragment,
    sportIdMatchupTypeInterface: SportIdMatchupTypeInterface
) : RecyclerView.Adapter<MenuAdapter.MyViewHolder>() {
    var objfragment: Fragment
    var SelectedIndex = 0
    var sportIdMatchupTypeInterface: SportIdMatchupTypeInterface

    inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener {
        var menuLayout: LinearLayout
        var txtMenu: TextView
        var imgMenuIcon: AppCompatImageView
        fun setClickListener(itemClickListener: OnItemClickListener?) {
            setClickListener(itemClickListener) // = itemClickListener;
        }

        override fun onClick(v: View) {
            println("testonclick")
        }

        //        private ItemClickListener clickListener;
        init {
            menuLayout = view.findViewById<View>(R.id.menuLayout) as LinearLayout
            txtMenu = view.findViewById<View>(R.id.txtMenu) as TextView
            imgMenuIcon = view.findViewById<View>(R.id.imgMenuIcon) as AppCompatImageView
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemView = LayoutInflater.from(parent.context)
            .inflate(R.layout.menu_itemview, parent, false)
        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val menulist = menuArrayList[position]
        println("jira menuadapter onBindViewHolder ${menulist.strName}" +
                " position $position")
        holder.txtMenu.text = menulist.strName

        if(SportAndIcon.map.containsKey(menulist.strName)) {
            SportAndIcon.map.get(menulist.strName)
                ?.let { holder.imgMenuIcon.setBackgroundResource(it) }
        }
        if (position == SelectedIndex) {
            holder.menuLayout.setBackgroundResource(R.drawable.corner_radius_black)
            holder.txtMenu.setTextColor(Color.WHITE)
          //  holder.imgMenuIcon.setImageResource(R.drawable.menu_selected)
            println("jira menuadapter SelectedIndex $SelectedIndex")
            holder.imgMenuIcon.supportBackgroundTintList = ContextCompat.getColorStateList(mContext, R.color.ball_selected_color)
           // holder.imgMenuIcon.setColorFilter(ContextCompat.getColor(mContext, R.color.white), android.graphics.PorterDuff.Mode.MULTIPLY);
        } else {
            println("jira menuadapter SelectedIndex ")

            holder.menuLayout.setBackgroundColor(Color.WHITE)
            holder.txtMenu.setTextColor(Color.BLACK)
          //  holder.imgMenuIcon.setImageResource(R.drawable.menu_unselected)
            holder.imgMenuIcon.supportBackgroundTintList = ContextCompat.getColorStateList(mContext, R.color.ball_default_color)

            //  holder.imgMenuIcon.setColorFilter(ContextCompat.getColor(mContext, R.color.ball_default_color), android.graphics.PorterDuff.Mode.MULTIPLY);

        }
        holder.menuLayout.setOnClickListener {
            SelectedIndex = position
            notifyDataSetChanged()
            if (position == 0) {
                (objfragment as HomeFragment).refreshSubMenu(menulist.arrLeague)
                sportIdMatchupTypeInterface.setSportAndMatchupType(-1, -1)
            } else {
                val leagues = ArrayList<Leagues?>()
                val objLeague = Leagues()
                objLeague.setLeagueAbbreviation(menulist.strName)
                leagues.add(0, objLeague)
                leagues.addAll(menulist.arrLeague)
                (objfragment as HomeFragment).refreshSubMenu(leagues)
                sportIdMatchupTypeInterface.setSportAndMatchupType(
                    menulist.arrLeague[0].sportId,
                    -1
                )
            }
        }
    }

    override fun getItemCount(): Int {
        println("jira menuadapter getItemCount ${menuArrayList.size}")
        return menuArrayList.size
    }

    init {
        objfragment = fragment
        this.sportIdMatchupTypeInterface = sportIdMatchupTypeInterface
    }

    fun resetSportTabColor()
    {

    }

    fun setSportTabBackgroundSelected()
    {

    }
}